### ISSUE TITLE
[v9] Unable to push form 

### ISSUE THREAD

**Issue by AaronSadlerUK** [KevinJump/uSync.Forms#6]
When I choose the "Push Items..." button for the selected form in the context menu, I get the below error on the remote server:

Umbraco: 9.0.1
uSync Complete: 9.0.1
uSync.Forms: 9.0.0

```
System.NullReferenceException: Unable to get Form with ID: 'Website Enquiry' off DB or from the cache.
   at Umbraco.Forms.Core.Services.FormService.Get(String formDataName)
   at uSync.Forms.Services.SyncFormService.GetForm(String name)
   at uSync.Forms.Serializers.FormSerializer.FindItem(String alias)
   at uSync.Core.Serialization.SyncSerializerRoot`1.FindItem(XElement node)
   at uSync.BackOffice.SyncHandlers.SyncHandlerRoot`2.SerializeFromNode(XElement node, SyncSerializerOptions options)
   at uSync.BackOffice.SyncHandlers.SyncHandlerRoot`2.IsItemCurrent(XElement node, SyncSerializerOptions options)
   at uSync.BackOffice.SyncHandlers.SyncHandlerRoot`2.ReportElement(XElement node, String filename, HandlerSettings settings, uSyncImportOptions options)
   at uSync.BackOffice.uSyncService.ReportPartial(String folder, uSyncPagedImportOptions options, Int32& total)
   at uSync.Expansions.Core.Services.SyncPackService.ReportChanges(SyncPackRequest request)
   at uSync.Publisher.Services.SyncPublishService.ReportChanges(SyncPublishRequest request)
   at uSync.Publisher.Controllers.uSyncReceiveApiController.DoPublisherStep(PublisherActionRequest options, Func`2 action)
   at uSync.Publisher.Controllers.uSyncReceiveApiController.ReportPackFolder(PublisherActionRequest options)
   at lambda_method66(Closure , Object , Object[] )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Umbraco.Cms.Web.Common.Middleware.BasicAuthenticationMiddleware.InvokeAsync(HttpContext context, RequestDelegate next)
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass6_1.<<UseMiddlewareInterface>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at Umbraco.Cms.Web.BackOffice.Middleware.BackOfficeExternalLoginProviderErrorMiddleware.InvokeAsync(HttpContext context, RequestDelegate next)
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass6_1.<<UseMiddlewareInterface>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Builder.Extensions.MapMiddleware.Invoke(HttpContext context)
   at Skybrud.Umbraco.Redirects.Middleware.RedirectsMiddleware.InvokeAsync(HttpContext context)
   at Microsoft.AspNetCore.Session.SessionMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Session.SessionMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Localization.RequestLocalizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authorization.Policy.AuthorizationMiddlewareResultHandler.HandleAsync(RequestDelegate next, HttpContext context, AuthorizationPolicy policy, PolicyAuthorizationResult authorizeResult)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at SixLabors.ImageSharp.Web.Middleware.ImageSharpMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.StatusCodePagesMiddleware.Invoke(HttpContext context)
   at StackExchange.Profiling.MiniProfilerMiddleware.Invoke(HttpContext context) in C:\projects\dotnet\src\MiniProfiler.AspNetCore\MiniProfilerMiddleware.cs:line 121
   at Umbraco.Cms.Web.Common.Middleware.UmbracoRequestMiddleware.InvokeAsync(HttpContext context, RequestDelegate next)
   at Umbraco.Cms.Web.Common.Middleware.UmbracoRequestMiddleware.InvokeAsync(HttpContext context, RequestDelegate next)
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass6_1.<<UseMiddlewareInterface>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at Umbraco.Cms.Web.Common.Middleware.PreviewAuthenticationMiddleware.InvokeAsync(HttpContext context, RequestDelegate next)
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass6_1.<<UseMiddlewareInterface>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at Umbraco.Cms.Web.Common.Middleware.UmbracoRequestLoggingMiddleware.InvokeAsync(HttpContext context, RequestDelegate next)
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass6_1.<<UseMiddlewareInterface>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Server.IIS.Core.IISHttpContextOfT`1.ProcessRequestAsync()
```

**Comment by KevinJump** [comment]
Grrr arggg forms - it has a habit of throwing errors when it can't find something .  fairly sure this particular one doesn't throw on v8

**Comment by AaronSadlerUK** [comment]
Interestingly the config file is on the server, I tried to push via Publisher AND via the traditional uSync importer both with the same issue

**Comment by Momkay** [comment]
Any updates on this issue? This has been een hassle lately when using multiple machines when working on the same project.

**Comment by AaronSadlerUK** [comment]
The nightly build resolved this for me

**Comment by KevinJump** [comment]
I will just confirm we've actually released that as a version

**Comment by Momkay** [comment]
> I will just confirm we've actually released that as a version

Am I missing something? I can only find the 9.0.0 october 12th version; https://www.nuget.org/packages/uSync.Forms

Thanks for the quick response.

**Comment by AaronSadlerUK** [comment]
@Momkay as a quick fix, the 9.0.1 nightly is on this feed:

https://pkgs.dev.azure.com/jumoo/Public/_packaging/nightly/nuget/v3/index.json

I'm sure @KevinJump will release it to nuget soon

**Comment by KevinJump** [comment]
Hi yeah - sorry looks like we never pushed that fix live. it is done now 9.0.1 is on nuget
